src/documentation/SearchResultList.tsx

```diff
@@ -0,0 +1,95 @@
+import { Divider, Link, Stack, Text } from "@chakra-ui/react";
+import { useRouterState } from "../router-hooks";
+import { Extract } from "./extracts";
+import { Result } from "./search-hooks";
+
+interface SearchResultListProps {
+  title: string;
+  results: Result[];
+  onClose: () => void;
+}
+
+const SearchResultList = ({
+  title,
+  results,
+  onClose,
+}: SearchResultListProps) => {
+  return (
+    <Stack spacing={2}>
+      <Text as="h2" fontSize="sm" px={3} color="gray.600" fontWeight="bold">
+        {title}
+      </Text>
+      {results.map((result) => (
+        <SearchResultItem key={result.id} value={result} onClose={onClose} />
+      ))}
+      {results.length === 0 && (
+        <Text as="h2" fontSize="sm" px={8}>
+          No results
+        </Text>
+      )}
+    </Stack>
+  );
+};
+
+interface SearchResultItemProps {
+  value: Result;
+  onClose: () => void;
+}
+
+const SearchResultItem = ({
+  value: { extract, navigation, containerTitle, title },
+  onClose,
+}: SearchResultItemProps) => {
+  const [, setState] = useRouterState();
+  return (
+    <Stack>
+      <Stack px={8} py={2} spacing={0}>
+        {title !== containerTitle && (
+          <Text fontSize="sm" color="gray.600" fontWeight="bold">
+            {containerTitle}
+          </Text>
+        )}
+        <Link
+          onClick={(e) => {
+            e.preventDefault();
+            onClose();
+            setState(navigation);
+          }}
+        >
+          <ExtractText extract={extract.title} title={true} />
+        </Link>
+        <ExtractText extract={extract.content} title={false} />
+      </Stack>
+      <Divider borderWidth="1px" color="gray.400" />
+    </Stack>
+  );
+};
+
+interface ExtractTextProps {
+  extract: Extract[];
+  title: boolean;
+}
+
+const ExtractText = ({ extract, title }: ExtractTextProps) => {
```

Comment:
I think it'd be better for ExtractTextProps to extend TextProps and spread them, rather than repeatedly switching on title. Then it'd be more direct styling of just the relevant Extract.

Reply:
@microbit-robert I think this is the last thing to tweak.

Reply:
[`4765f1e`](https://github.com/microbit-foundation/python-editor-next/pull/452/commits/4765f1e6ff1cbe218ea85e7d4f2493c7f622c5f6)